Tutorial on background linters

expe opened the pull request:
A start on the 'as you type' background linters tutorial. Let me know if it's easy to follow of if it's confusing as hell.

Reference to issue:
https://github.com/Financial-Times/ft-origami/issues/399

AlbertoElias commented on docs/developer-guide/background-linters.md:
```diff
@@ -0,0 +1,86 @@
+
+If you already have Origami Build Tools (OBT) and ESLint configured, skip to the [Installing background linter plugins](#Installing-background-linter-plugins) section.
+
+Ensure you have installed [Origami Build Tools](https://www.npmjs.com/package/origami-build-tools). This will provide you with Sass, SCSS-lint and provide instructions on how to set up Node.js and Ruby. 
```
Scss-Lint

AlbertoElias replied:
Actually, better to go all lower case. The thing is, it's Scss and Sass, but the guys behind scss-lint write Sass wrong :P

triblondon commented on docs/developer-guide/background-linters.md:
```diff
@@ -0,0 +1,86 @@
+
+# Setting up background linters
+
+ESLint *must* be used when developing Origami components. It's useful to have linters run in the background and highlight errors as you're typing. SublimeLinter and ESLint packages are recommended for Sublime Text and JetBrains IDEs respectively.
```
This is probably a *should*.

triblondon commented on docs/developer-guide/background-linters.md:
```diff
@@ -0,0 +1,86 @@
+
+ESLint isn't installed automatically installed as part of OBT, so install it with the following command:
+
+**Enter the following to your shell**
```
Should this be #### rather than **?

expe commented:
* Added submenu entry under developer guide
* Changed must to should
* Added #### headings
* Changed code block to tab indentation. Techdocs seems to break the triple backtick.

triblondon commented on docs/developer-guide/background-linters.md:
```diff
@@ -0,0 +1,84 @@
+ESLint isn't installed automatically installed as part of OBT, so install it with the following command:
+
+#### Enter the following to your shell
+`npm install -g eslint`
```
There's a techdocs style for input/output, can you use that?

triblondon commented:
one further tweak than feel free to merge.
